es/es-core/src/main/java/org/fao/geonet/es/EsClient.java

```diff
@@ -349,12 +258,12 @@ public static String analyzeField(String collection,
                         return token.get("token").getAsString();
                     }
                 }
-                return fieldValue;
+                return "";
             } else {
-                return fieldValue;
+                return "";
             }
         } catch (Exception ex) {
-            return fieldValue;
+            return "";
```

Comment:
The three '''return "";''' should be replaced by one and comment in javadoc should be updated.
This said, maybe converting result to json before to process them is of no use... But I do not understand what the purpose of the method is. A unit test should be good way to document it.
Finally, as method method use "EsClient.get().getClient()", one could also imagine analyseField NOT to be static, and that one have first to perfom EsClient.get() to use analyseField.

Reply:
> The three '''return "";''' should be replaced by one and comment in javadoc should be updated.
 
Changed.

> This said, maybe converting result to json before to process them is of no use... 

To get the type property I don't see other option ?

![image](https://user-images.githubusercontent.com/1701393/48128223-17f18780-e286-11e8-8d8c-ebe34cd8c376.png)

> A unit test should be good way to document it.

Not sure how to start ES to get the Analyze API available for this. But there is probably ways to achieve this. I would say this is something to cover in the `es` branch unless you have any guideline for this? 


> Finally, as method method use "EsClient.get().getClient()", one could also imagine analyseField NOT to be static, and that one have first to perfom EsClient.get() to use analyseField.

This is called from XSLT, so it has to be static but maybe I did not get it ?

Reply:
static call are possible, e.g.
https://github.com/geonetwork/core-geonetwork/pull/2847/commits/d529b037c441696c380e088c60fb6c401561bd6c#diff-795afefc5ecb4cd7395b302e616d117eR38

Reply:
I thought we were talking of es branch !

Reply:
Ok was not aware of that. I don't know why we took the habit to do that.

Reply:
> I thought we were talking of es branch !

We are not :) I've 2 projects using master and Elsaticsearch mainly for making dashboards.

Reply:
Not using JEST. At some point in ES branch we should check if it is better to move to ES Java API instead which could be better for this kind of things (but for now we can't use it due to the version of Lucene used by GN).